Tests/MongoSwiftTests/SpecTestRunner/ChangeStreamSpecTest.swift

```diff
@@ -0,0 +1,136 @@
+            let key = elem.key
+            return key != "_id"
+        }
+
+        return lhsFullDoc == rhsFullDoc && lhs.ns == rhs.ns && lhs.operationType == rhs.operationType
+    }
+}
+
+internal protocol ChangeStreamSpecTest: Decodable {
+    var description: String { get }
+    var operations: [ChangeStreamAnyTestOperation] { get }
+    var result: TestResult { get }
+
+    func run(client: MongoClient,
+             seenError: Error?,
+             changeStream: ChangeStream<ChangeStreamTestEventDocument>?) throws
+}
+
+extension ChangeStreamSpecTest {
+    func run(client: MongoClient,
+             seenError: Error?,
+             changeStream: ChangeStream<ChangeStreamTestEventDocument>?) throws {
+        var seenError = seenError
+        for operation in self.operations {
+            guard let dbName = operation.database else {
+                print("The database name for running the operation is missing.")
+                return
+            }
+
+            guard let collName = operation.collection else {
+                print("The collection name for running the operation is missing.")
+                return
+            }
+
+            let db = client.db(dbName)
+            let coll = db.collection(collName)
+
+            do {
+                try operation.anyTestOperation.op.execute(client: client,
+                                                          database: db,
+                                                          collection: coll,
+                                                          session: nil)
+            } catch {
+                seenError = error
+            }
+        }
+
+        // If there was no error and result.error is set, iterate changeStream once and capture any error.
+        if seenError == nil && result.error != nil {
+            do {
+                try changeStream?.nextOrError()
+            } catch {
+                seenError = error
+            }
+        }
+
+        if seenError != nil {
+            // assert errors match expected errors
+            assertError(seenError: seenError)
+        } else {
+            // assert change doc match expected change doc
+            assertSuccess(changeStream: changeStream)
+        }
+     }
+
+     private func assertError(seenError: Error?) {
+        // assert that an error was expected for the test
+        expect(self.result.error).toNot(beNil())
+        // TODO: assert that the error matches expected error
+    }
+
+    private func assertSuccess(changeStream: ChangeStream<ChangeStreamTestEventDocument>?) {
+        // assert that no error was expected for the test
+        expect(self.result.error).to(beNil())
+        // assert that the changes received from changeStream MATCH the results in result.success
+        if let changeStream = changeStream, let expectedChanges = self.result.success {
+            for expectedChange in expectedChanges {
+                if let change = changeStream.next() {
```

Comment:
we want to fail the test if this `if let` doesn't succeed, so maybe rewrite it as:
```swift
let change = changeStream.next()
expect(changeStream.error).to(beNil(), description: self.description)
expect(change).toNot(beNil(), description: self.description)
expect(change!).to(equal(expectedChange), description: self.description)
```

As another note: all the `expect` calls in this test should have `self.description` as the description so that we know which individual test failed when looking at failed assertions.